codegen/src/lib.rs

```diff
@@ -40,7 +40,7 @@ pub use syn;
 ///
 /// Generating an interface using all of the defaults:
 ///
-/// ```rust
+/// ```rust,standalone_crate
```

Comment:
Doc tests have been hanging in CI, and even locally, I think since I enabled Rust 2024. Rust 2024 changes how doc tests run and compiles them into a single binary ratehr than 1 per test.

Since many of our doc tests generate a load of code via the `subxt` macro, I suspect that this was eating up too much memory during compilation and causing the CI machine to hang (locally it went well over 4GB offhand).

Either way, adding `standalone_crate` to them all reverted to the previous behaviour which seems to work much better for us!